agent/xds/server.go

```diff
@@ -141,6 +150,36 @@ type Server struct {
 	AuthCheckFrequency time.Duration
 
 	DisableV2Protocol bool
+
+	activeStreams activeStreamCounters
+}
+
+// activeStreamCounters simply encapsulates two counters accessed atomically to
+// ensure alignment is correct.
+type activeStreamCounters struct {
+	xDSv3 uint64
+	xDSv2 uint64
+}
+
+func (c *activeStreamCounters) Increment(xdsVersion string) func() {
+	var counter *uint64
+	switch xdsVersion {
+	case "v3":
+		counter = &c.xDSv3
+	case "v2":
+		counter = &c.xDSv2
+	default:
+		return func() {}
+	}
+
+	count := atomic.AddUint64(counter, 1)
+	metrics.SetGaugeWithLabels([]string{"xds", "server", "streams"}, float32(count),
+		[]metrics.Label{{Name: "version", Value: xdsVersion}})
+	return func() {
+		count := atomic.AddUint64(counter, ^uint64(0))
+		metrics.SetGaugeWithLabels([]string{"xds", "server", "streams"}, float32(count),
+			[]metrics.Label{{Name: "version", Value: xdsVersion}})
```

Comment:
Minor: could define the labels once and use them in both calls to set gauges to shrink the line lengths a bit.

Reply:
:+1: